apps/examples/testcase/le_tc/kernel/tc_libc_misc.c

```diff
@@ -347,6 +348,76 @@ static void tc_libc_misc_vdbg(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @fn                  :tc_libc_misc_dumpbuffer
+ * @brief               :Do a pretty buffer dump
+ * @scenario            :Do a pretty buffer dump
+ * @API's covered       :lib_dumpbuffer
+ * @Preconditions       :None
+ * @Postconditions      :None
+ * @Return              :void
+ */
+static void tc_libc_misc_dumpbuffer(void)
+{
+	const char *msg = "tc_libc_misc_dumpbuffer";
+	unsigned char buffer[] = { 'S', 'A', 'M', 'S', 'U', 'N', 'G', '-', 'T', 'i', 'z', 'e', 'n', 'R', 'T' };
+	unsigned char *buf = NULL;
+	int ret_chk;
+	int idx;
+
+	/**
+	 * As lib_dumpbuffer returns void, there in no way to check the success or failure case
+	 * we need to manually check the dump output
+	 * Case-1: output: 53414d53554e472d54697a656e5254 SAMSUNG-TizenRT
+	 * Case-2: output: 0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a ............... (i.e, 256 "0a")
+	 */
+
+	/* Case-1 */
+	lib_dumpbuffer(msg, buffer, sizeof(buffer));
+
+	/* Case-2 */
+	buf = (unsigned char *)malloc(BUFF_SIZE);
+	if (!buf) {
+		ret_chk = dbg("malloc failed");
+		TC_ASSERT_EQ("dbg", ret_chk, strlen("malloc failed"));
```

Comment:
I think that it seems to be unnecessary.
Why do you check whether malloc is failed or not using dbg in this tc?
TC_ASSERT("xxx", buf) is enough